lib/sles4sap/sap_deployment_automation_framework/basetest.pm

```diff
@@ -27,6 +30,14 @@ sub post_fail_hook {
     az_login();
     sdaf_cleanup();
     disconnect_target_from_serial();
+
+    # Cleanup VM resources
```

Comment:
could you please elaborate on this comment? I feel it is an important one but I do not fully  get the point.
It is about deleting VM and not all the resource_group. Which VM in particular? The deployer one? Not HANA or any SAP workload related one, right?
And why we delete deployer VM? ... or better to say: what is the lifecycle of that after this PR?

Reply:
The process is:
1. test starts
2. clone of original deployer VM is created inside permanent 'OpenQA_SDAF_DEPLOYER' RG
3. this cloned VM is only temporary and is destroyed after test finished
4. Resource group 'OpenQA_SDAF_DEPLOYER' must not be destroyed

This means cleanup goes through the RG and destroys only resources specific to the test. 

I have updated the comment, but my next ticket is about improving cleanup so I will address it there.